pkg/controller/cainjector/sources.go

```diff
@@ -87,7 +87,7 @@ func (c *certificateDataSource) ReadCA(ctx context.Context, log logr.Logger, met
 	certName := splitNamespacedName(certNameRaw)
 	log = log.WithValues("certificate", certName)
 	if certName.Namespace == "" {
-		log.Error(nil, "invalid certificate name; needs a namespace/ prefix")
+		log.Error(nil, "invalid certificate name: needs a namespace/ prefix")
```

Comment:
I think it would be better to supply an error rather than nil. Could be `errors.New("invalid annotation")`  or something.
Otherwise the logged message says `error=nil` which is weird.